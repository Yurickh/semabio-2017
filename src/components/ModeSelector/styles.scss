@import 'ambience/color';

.mode-selector {
  display: flex;
  justify-content: center;

  & > .mode {
    font-size: 20px;

    margin-right: 42px;

    cursor: pointer;
    letter-spacing: .71px;

    color: $color-light;
  }

  & > .mode.-selected {
    color: #000;
  }
}
