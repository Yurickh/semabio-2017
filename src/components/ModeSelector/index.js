import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './styles.css'

class ModeSelector extends Component {
	static propTypes = {
		currentMode: PropTypes.number,
		onChange: PropTypes.func,
	}

	isSelected = (index) => {
		return this.props.currentMode === index ? '-selected' : ''
	}

	selectMode = (index) => {
		this.props.onChange(index)
	}

	render() {
		return (
			<div className="mode-selector">
				{
					['Minicursos', 'Palestras', 'Rodas de conversa'].map((mode, index) => {
						return <div
							className={`mode ${this.isSelected(index)}`}
							onClick={() => this.selectMode(index)}
							key={mode}
						>
							{mode}
						</div>
					})
				}
			</div>
		)
	}
}

export default ModeSelector
