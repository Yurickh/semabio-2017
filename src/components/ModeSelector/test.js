import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import ModeSelector from '../ModeSelector'

describe('ModeSelector', () => {
  it('should render correctly', () => {
    const tree = shallow(<ModeSelector />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
