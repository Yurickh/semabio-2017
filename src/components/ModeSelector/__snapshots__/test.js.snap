// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeSelector should render correctly 1`] = `
<div
  className="mode-selector"
>
  <div
    className="mode "
    onClick={[Function]}
  >
    Minicursos
  </div>
  <div
    className="mode "
    onClick={[Function]}
  >
    Palestras
  </div>
  <div
    className="mode "
    onClick={[Function]}
  >
    Rodas de conversa
  </div>
</div>
`;
